GoogleForCreators/web-stories-wp
# I18N: Fix loading of JavaScript translations

## Context



As per #8994, JavaScript translations are currently not being loaded.

## Summary



This PR ensures that chunks from dynamic imports are registered as scripts in WordPress, without actually enqueueing them. The chunks' translations will be printed whenever the main script is enqueued.

## Relevant Technical Choices



* Add missing `wp_set_script_translations()` call for chunks
* Register chunks from dynamic imports as scripts and print their translations whenever the main script is enqueued.

## To-do



- As per discussion below, some more predictable file names might be valuable.

## User-facing changes



After manually renaming some translation files as per below instructions, the UI is localized again:

![Screenshot 2021-09-10 at 11 44 28](https://user-images.githubusercontent.com/841956/132834829-549e629b-6734-40d8-8125-f3f2b5b1ba0d.png)


## Testing Instructions






While it's technically a user-facing change (UI should be translated again), it's not currently easily possible to test this change.

Why?

To implement JavaScript localization, WordPress extracts translatable strings from all JavaScript files in a plugin. For each JavaScript file, there is 1 corresponding translation file, which contains the MD5 hash of the JS file path.

The problem is that our JS files have hashes in their file names.

So when you download translations from WordPress.org, the translation file name will be `web-stories-de_DE-<hash of the JS file from v1.11.0>.json`, but the hash will be different when building the plugin from `main`. `web-stories-de_DE-<hash of the JS file from main branch>.json` does not exist of course.

The only way to test this is by manually renaming some of the JSON translation file names in `wp-content/languages/plugins` with updated MD5 hashes.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #8994
